### ISSUE TITLE
fwrite doesn't save dates

### ISSUE THREAD

**Issue by skanskan** [Rdatatable/data.table#1772]
Hello.

I have a dataset.  I can choose to load it on R from a Stata file or from a SPSS file.
In both cases it's loaded properly with the _haven_ package.
The dates are recognized properly.

But when I save it to disk with data.table's fwrite.
`fwrite(ppp, "ppp.csv",  sep=",", col.names = TRUE) `
I have a problem, the dates dissapear and are converted to numbers with no sense.

For example the date 1967-08-06  now is  -879

I've also tried playing with fwrite options, such as quote=FALSE, with no success.

I've uploaded a small sample of the files, the spss, the stata and the saved csv.

```
library(haven)
library(data.table)
ppp <- read_sav("pspss.sav") # choose one of these two.
ppp <- read_dta("pstata.dta")  # choose one of these two.
fwrite(ppp, "ppp.csv",  sep=",", col.names = TRUE) 
```

http://www73.zippyshare.com/v/OwzwbyQq/file.html

How can I solve it?

Regards

**Comment by arunsrinivasan** [comment]
Duplicate of #1664. It's not yet implemented. 

What you're seeing is the internal representation. By just setting attribute to `Date`, you should be able to see the date.

**Comment by skanskan** [comment]
When should I set the class attribute to Date? When fwriting or when reading back? And how?

Regards

**Comment by arunsrinivasan** [comment]
After reading it back. `attr(dt$col) <- "Date"`.

**Comment by skanskan** [comment]
OK, thank you.
Any easy way to do it easily?.  The real file has 1200 variables, many of them are dates.
I mean some way to save the original classes and change the new...

**Comment by skanskan** [comment]
Since it seems there is no simple solution I'm trying to store column classes and change them back again.

I've taken the original dataset ppp, 
  
    areDates <- (sapply(ppp, class) == "Date")

I save it on an file and I can read it next time.

    ppp <- fread("ppp.csv", encoding="UTF-8")

And now I change the classes of the newly read dataset back to the original one.   

    ppp[,names(ppp)[areDates] := lapply(.SD,as.Date) , .SDcols = areDates ]

The problem is that it works with this toy example but with the real big example I get the error:

> Error in as.Date.numeric(X[[i]], ...) : 'origin' must be supplied

I've also tried with
     lapply(.SD,function(xx) class(xx) <- "Date")
but it fills the variables with NA

**Comment by jangorecki** [comment]
@skanskan I would try `lapply(., setattr, "class", "Date")`, but just guessing. Changing class of elements in a list fits more on Stackoverflow. Date class is just a numeric having own methods, you "convert" it just by add or drop class attribute. There is separate issue which discuss that fact and its potential to IO performance (fread-fwrite) when writing Date in string representation will be implemented, see #1656
